# small RNAseq pipelines - current status

Created: 2020/09/24 15:19:42
Last modified: 2021/08/03 10:57:21

- **Aim:** Evaluate the current pipelines available for processing **small RNA-seq** data

## Table of contents

- [small RNAseq pipelines - current status](#small-rnaseq-pipelines---current-status)
  - [Table of contents](#table-of-contents)
  - [Pipelines](#pipelines)
  - [Other things of possible interest](#other-things-of-possible-interest)
  - [Notes](#notes)

## Pipelines

See [small RNAseq pipelines](./small_RNAseq_pipelines.csv) for a spreadsheet of pipelines

## Other things of possible interest

- Prediction and annotation of tRNA-derived small RNAs [here](https://github.com/wangqinhu/tsRFinder)
- A JBrowse plugin to support small RNA visualization [here](https://github.com/bhofmei/jbplugin-smallrna)
- A set of tools related to the Rfam (database of non-coding RNA families https://rfam.org/) production pipeline [here](https://github.com/Rfam/rfam-production)
- Script to map annotated ncRNAs (and other elements) to a chromosome and visualise [here](https://github.com/fanagislab/draw_annotation/tree/master/bin)
- [HPC-REDItools - A tool for large-scale RNA-editing analysis](https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/s12859-020-03562-x) (RNA-editing is a molecular process through which some cells can make discrete changes to specific nucleotide sequences within an RNA molecule after it has been generated by RNA polymerase)

## Notes

- [smrnaseq](https://github.com/nf-core/smrnaseq) is still the most robust pipeline Iâ€™ve come across in terms of:
  - being in a workflow language
  - being specific to both small and non-coding RNA
  - being deployable to cluster/cloud environments
- There are far more pipelines available for RNA-seq data
- There seems to very few pipelines dedicated to both small and non-coding RNA-seq
