# small RNAseq pipelines - current status

Created: 2020/09/24 15:19:42
Last modified: 2021/07/26 17:53:41

- **Aim:** Evaluate the current pipelines available for processing **small RNA-seq** data

## Table of contents

- [small RNAseq pipelines - current status](#small-rnaseq-pipelines---current-status)
  - [Table of contents](#table-of-contents)
  - [Pipelines](#pipelines)
  - [Other things of possible interest](#other-things-of-possible-interest)
  - [Notes](#notes)

## Pipelines

See [small RNAseq pipelines](./small_RNAseq_pipelines.csv)

## Other things of possible interest

- Prediction and annotation of tRNA-derived small RNAs [here](https://github.com/wangqinhu/tsRFinder)
- A JBrowse plugin to support small RNA visualization [here](https://github.com/bhofmei/jbplugin-smallrna)
- A set of tools related to the Rfam (database of non-coding RNA families https://rfam.org/) production pipeline [here](https://github.com/Rfam/rfam-production)
- Script to map annotated ncRNAs (and other elements) to a chromosome and visualise [here](https://github.com/fanagislab/draw_annotation/tree/master/bin)
- [HPC-REDItools - A tool for large-scale RNA-editing analysis](https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/s12859-020-03562-x) (RNA-editing is a molecular process through which some cells can make discrete changes to specific nucleotide sequences within an RNA molecule after it has been generated by RNA polymerase)

## Notes

- [smrnaseq](https://github.com/nf-core/smrnaseq) is still the most robust pipeline I’ve come across in terms of:
  - being in a workflow language
  - being specific to both small and non-coding RNA
  - being deployable to cluster/cloud environments
- There are far more pipelines available for RNA-seq data
- There seems to very few pipelines dedicated to both small and non-coding RNA-seq
- There are a bunch of interesting scripts available that might end up being useful to incorporate or compliment whichever pipeline we end up using, or be used for future analyses:
  - [CloseCall](https://github.com/StevenWingett/CloseCall) can identify RNA-RNA interactions
  - [DiMSum](https://github.com/lehner-lab/DiMSum) uses deep mutational scanning to enable the multiplexed measurement of the effects of thousands of variants of proteins, RNAs, and regulatory elements
  - [HPC-REDItools](https://bmcbioinformatics.biomedcentral.com/track/pdf/10.1186/s12859-020-03562-x) is a tool for large-scale RNA-editing analysis
  - [draw_annotation](https://github.com/fanagislab/draw_annotation) to map annotated ncRNAs (and other elements) to a chromosome and visualise
- There are some other interesting pipelines:
  - [sRNAflow](https://github.com/zajakin/sRNAflow) is a Nextflow pipeline to analyse small RNA that is tailored to meet the specific needs of samples derived from biofluids
  - [exceRNApipeline](https://github.com/zhuchcn/exceRNApipeline) is a Snakemake pipeline for extracellular small RNA-seq
- There are a few interesting annotation pipelines:
  - [FlaiMapper](https://github.com/yhoogstrate/flaimapper )
  - [short-ncrna-annotation](https://github.com/SimonSchafferer/short-ncrna-annotation) 
  - However, these are typically older repos, and the scripts aren’t wrapped in a workflow language
  - Additionally it looks like there is some overlap in the analyses in smrnaseq and FlaiMapper 
- Some interesting and more general RNA-Seq pipelines:
  - [viper-rnaseq](https://github.com/hanfeisun/viper-rnaseq)
  - [rna-seq-star-deseq2](https://github.com/snakemake-workflows/rna-seq-star-deseq2)
  - [rna-seq-kallisto-sleuth](https://github.com/snakemake-workflows/rna-seq-kallisto-sleuth)
  - [RNA-seq](https://github.com/biowdl/RNA-seq)
  - [snakemake_deeptools](https://github.com/AngryMaciek/snakemake_deeptools)
  - [omics_pipe](https://pypi.org/project/omics_pipe/)
